tools/configuration.py

```diff
@@ -71,49 +83,27 @@ def _check_pre_reqs(self):
         if not self.get_gnuplot_version():
             raise ConfigurationException("gnuplot not found")
 
-    def process_and_validate_params(self, args_orig):
+    def _process_and_validate_params(self, args_orig=None):
         self._check_pre_reqs()
 
-        optlist, args = getopt.getopt(args_orig, 'hc:', ["help"])
-        result_opt = {}
-        for o, v in optlist:
-            if o == '-c':
-                key, value = v.split('=', 1)
-                if key not in self.conf:
-                    raise KeyError('no such key "%s" in config' % key)
-                result_opt[key] = value
-                if isinstance(self.conf[key], int):
-                    self.conf[key] = int(value)
-                else:
-                    self.conf[key] = value
-            elif o in ('-h', '--help'):
-                raise UsageException()
-
-        if len(args) < 2:
-            raise UsageException("Too little args")
-
-        if not self.is_csv_output() and not self.is_html_output():
-            raise UsageException(format('Invalid output parameter: %s' % self.conf['output']))
-
-        outputpath = os.path.abspath(args[-1])
+        args = get_gitstat_parser().parse_args(args_orig)
+
         try:
-            os.makedirs(outputpath)
+            os.makedirs(args.output_path)
         except OSError:
             pass
-        if not os.path.isdir(outputpath):
+        if not os.path.isdir(args.output_path):
```

Comment:
check this: https://stackoverflow.com/questions/11415570/directory-path-types-with-argparse for inspiration how path can be validated